models.go

```diff
@@ -0,0 +1,25 @@
+package watermill_amazonsqs
+
+import (
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+)
+
+type Marshaler interface {
+	Marshal(msg *message.Message) (*sqs.Message, error)
+}
+
+type Unmarshaler interface {
+	Unmarshal(msg *sqs.Message) (*message.Message, error)
+}
+
+type PublisherConfig struct {
```

Comment:
Putting these structs outside seems like we are forcing separation when it may be not needed.
I'm not sure yet, but I'm thinking how to keep the packages.

One way is to put all code in `aws` package (`watermill-aws/aws`), with proper methods like `aws.NewSNSPublisher` and `aws.NewSQSPublisher`. We don't have to worry about "common" code then, but it can get more cluttered.

The other way is to keep it separate, so `watermill-aws/sns` and `watermill-aws/sqs` with methods like `sns.NewPublisher` and `sqs.NewPublisher`. Better separation, but still we will have some common code for sure.

Hard to say which approach is better for now, we can always refactor it later.

@roblaszczak @maclav3 any thoughts from you guys?

Reply:
I would like to see a the Adapter be "aws" with "sqs" and "sns" packages.  We could leave common structs and code in the "aws" package and the specific structs and code in the defined packages.  This might also allow for the creation of a "s3" package in the future to deal with S3 file flows.

Reply:
Cool, I like that idea. 👍 Another thing is whether SNS/SQS/S3 will use the same publisher config, because there can be some specific fields for each. But I think we can stick with common for now and refactor later if there's a need. :)

Reply:
Although the Marshaler/Unmarshaler pair specific for `sqs.Message`, so I guess we should have them separately? Also, we could just duplicate the Subscriber/Publisher configs for now, it's not a big cost and will allow for cleaner interface (you can use the same import package for config and for constructor).